tools/sylph/sylph.xml

```diff
@@ -0,0 +1,392 @@
+                #set $read1 = re.sub(r"\s+", "_", $sketch.input_1.element_identifier)
+            #else:
+                #set $read1 = re.sub(r"\s+", "_", str($sketch.input_1.element_identifier) + "." + str($ext_1))
+            #end if
+            #if $sketch.input_2.element_identifier.endswith(".fastq") or $sketch.input_2.element_identifier.endswith(".fastq.gz"):
+                #set $read2 = re.sub(r"\s+", "_", $sketch.input_2.element_identifier)
+            #else:
+                #set $read2 = re.sub(r"\s+", "_", str($sketch.input_2.element_identifier) + "." + str($ext_2))
+            #end if
+            ln -s "$sketch.input_1" "$read1" &&
+            ln -s "$sketch.input_2" "$read2" &&
+
+##Paired  group
+        #else if $sketch.type == "paired_group":
+            #if $sketch.input.forward.ext == "fastqsanger"
+                #set $ext_1 = "fastq"
+            #else if $sketch.input.forward.ext == "fastqsanger.gz":
+                #set $ext_1 = "fastq.gz"
+            #else:
+                #set $ext_1 = str($sketch.input.forward.ext)
+            #end if
+
+            #if $sketch.input.reverse.ext == "fastqsanger"
+                #set $ext_2 = "fastq"
+            #else if $sketch.input.reverse.ext == "fastqsanger.gz":
+                #set $ext_2 = "fastq.gz"
+            #else:
+                #set $ext_2 = str($sketch.input.reverse.ext)
+            #end if
+            
+            #set $read1 = re.sub(r"\s+", "_", str($sketch.input.element_identifier) + "." + str($ext_1))
+            #set $read2 = re.sub(r"\s+", "_", str($sketch.input.element_identifier) + "_r2." + str($ext_2))
+            ln -s "$sketch.input.forward" "$read1" &&
+            ln -s "$sketch.input.reverse" "$read2" &&
+        #end if
+##SKETCHING
+        sylph sketch
+        #if $sketch.type == "single":
+            $input
+        #else if $sketch.type == "single_group":
+            -r $input
+        #else if $sketch.type == "paired": 
+            -1 $read1
+            -2 $read2
+        #else if $sketch.type == "paired_group": 
+            -1 $read1
+            -2 $read2
+        #end if
+        -t \${GALAXY_SLOTS:-4}
+        -d sylph_sketches &&
+
+##OPTION FOR PROFILE
+        #if $function.functions == "profile"
+            sylph profile
+            input.syldb
+            sylph_sketches/*.sylsp
+            #if $function.min_num_kmers:
+                --min-number-kmers ${function.min_num_kmers}
+            #end if
+            -t \${GALAXY_SLOTS:-4}
+            -o output.tsv
+
+            #if $function.outputs:
+                #if $database_select.select == "cached":
+                    && ln -s "$database_select.sylph_database.fields.path/database.tsv.gz" "database.tsv.gz"
+                #else:
+                    && ln -s "$database_select.metadata" "database.tsv.gz"
+                #end if
+                && python "$__tool_directory__/sylph_to_taxprof.py" -s output.tsv -m database.tsv.gz -o metaphlan_
```

Comment:
You probably need to put in a `# if "metaphlan" in $function.outputs:`

Reply:
The metaphlan tool must be run regardless if krona, metphlan, or both are requested. The krona formatted file is generated from the metaphlan converted file. The #if part is about if we hand it back to the user